.github/workflows/pr-vrt-comment.yml

```diff
@@ -34,134 +31,104 @@ jobs:
       # downloaded artifacts will contain screenshots from affected project including 'screenshots-report.json' which contains proper image mappings for affected project
       # - see @{link file://./../scripts/prepare-vr-screenshots-for-upload.js#43}
       # - see @{link file://./pr-vrt.yml#56}
-      - uses: actions/download-artifact@v4
+
+      - name: Check for vrscreenshot artifact
+        id: check_artifact
+        uses: actions/github-script@v7
         with:
-          name: vrscreenshot
-          path: ./screenshots
-          run-id: ${{ github.event.workflow_run.id }}
-          github-token: ${{ secrets.GITHUB_TOKEN }}
+          script: |
+            const artifacts = await github.rest.actions.listWorkflowRunArtifacts({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              run_id: context.payload.workflow_run.id
+            });
+
+            const artifactExists = artifacts.data.artifacts.some(artifact => artifact.name === "vrscreenshot");
+
+            if (artifactExists) {
+              core.info("Artifact 'vrscreenshot' found.");
+              core.exportVariable("run_diff", "true");
+            } else {
+              core.info("Artifact 'vrscreenshot' not found.");
+              core.exportVariable("run_diff", "false");
+            }
 
       - uses: actions/download-artifact@v4
+        if: ${{ env.run_diff == 'true' }}
         with:
-          name: pr-number
-          path: ./results
+          name: vrscreenshot
+          path: ./screenshots
           run-id: ${{ github.event.workflow_run.id }}
           github-token: ${{ secrets.GITHUB_TOKEN }}
 
-      - name: Load PR number
+      - name: Save metadata
         uses: actions/github-script@v7
-        id: pr_number
+        if: ${{ env.run_diff == 'true' }}
         with:
           script: |
-            const run = require('./.github/scripts/validate-pr-number');
-            const result = run({filePath:'results/pr.txt'});
-            return result;
-          result-encoding: string
+            const fs = require('fs');
 
-      - name: VR App - Create Policy
-        run: |
-          echo "MAKE THIS STEP WORK"
+            const data = {
```

Comment:
we probably need to add validation of these dynamically created strings to avoid poisoning attack vector

Reply:
Note that uploading artifact with PR number and then downloading it is still vulnerable to attack, someone can change the artifact uploaded to a random PR number for spamming or giving incorrect results. String validation won't change that.

Compare that to directly accessing triggering workflow information, and we don't have to deal with any such issues, as this information is set by github directly with no possibility for an attacker to intervene. At most the attacker can change the source branch name. I will check how to mitigate that.

Reply:
I have moved the bash script to javascript which avoid scripts injection attacks.